### ISSUE TITLE
process.Name() returns truncated process name

### ISSUE THREAD

**Issue by jcaberio** [shirou/gopsutil#300]
Python version of psutil had this  [bug before](https://github.com/giampaolo/psutil/issues/628) but it's already fixed. Below are the python and golang versions of the test code. Notice that 'netcast-dashboard' was displayed correctly on python but truncated on golang psutil.

```
# pypsutil.py
import psutil

for proc in psutil.process_iter():
    try:
        pinfo = proc.as_dict(attrs=['pid', 'name'])
    except psutil.NoSuchProcess:
        pass
    else:
        print(pinfo)
```

```
$ python pypsutil.py
{'pid': 1, 'name': 'init'}
{'pid': 2, 'name': 'kthreadd'}
{'pid': 3, 'name': 'migration/0'}
{'pid': 4, 'name': 'ksoftirqd/0'}
{'pid': 5, 'name': 'migration/0'}
{'pid': 6, 'name': 'watchdog/0'}
{'pid': 7, 'name': 'migration/1'}
{'pid': 8, 'name': 'migration/1'}
{'pid': 9, 'name': 'ksoftirqd/1'}
{'pid': 10, 'name': 'watchdog/1'}
{'pid': 11, 'name': 'events/0'}
{'pid': 12, 'name': 'events/1'}
{'pid': 13, 'name': 'cgroup'}
{'pid': 14, 'name': 'khelper'}
{'pid': 15, 'name': 'netns'}
{'pid': 16, 'name': 'async/mgr'}
{'pid': 17, 'name': 'pm'}
{'pid': 18, 'name': 'sync_supers'}
{'pid': 19, 'name': 'bdi-default'}
{'pid': 20, 'name': 'kintegrityd/0'}
{'pid': 21, 'name': 'kintegrityd/1'}
{'pid': 22, 'name': 'kblockd/0'}
{'pid': 23, 'name': 'kblockd/1'}
{'pid': 24, 'name': 'kacpid'}
{'pid': 25, 'name': 'kacpi_notify'}
{'pid': 26, 'name': 'kacpi_hotplug'}
{'pid': 27, 'name': 'ata_aux'}
{'pid': 28, 'name': 'ata_sff/0'}
{'pid': 29, 'name': 'ata_sff/1'}
{'pid': 30, 'name': 'ksuspend_usbd'}
{'pid': 31, 'name': 'khubd'}
{'pid': 32, 'name': 'kseriod'}
{'pid': 33, 'name': 'md/0'}
{'pid': 34, 'name': 'md/1'}
{'pid': 35, 'name': 'md_misc/0'}
{'pid': 36, 'name': 'md_misc/1'}
{'pid': 37, 'name': 'linkwatch'}
{'pid': 38, 'name': 'khungtaskd'}
{'pid': 39, 'name': 'kswapd0'}
{'pid': 40, 'name': 'ksmd'}
{'pid': 41, 'name': 'khugepaged'}
{'pid': 42, 'name': 'aio/0'}
{'pid': 43, 'name': 'aio/1'}
{'pid': 44, 'name': 'crypto/0'}
{'pid': 45, 'name': 'crypto/1'}
{'pid': 50, 'name': 'kthrotld/0'}
{'pid': 51, 'name': 'kthrotld/1'}
{'pid': 52, 'name': 'pciehpd'}
{'pid': 54, 'name': 'kpsmoused'}
{'pid': 55, 'name': 'usbhid_resumer'}
{'pid': 85, 'name': 'kstriped'}
{'pid': 169, 'name': 'mpt_poll_0'}
{'pid': 170, 'name': 'mpt/0'}
{'pid': 171, 'name': 'scsi_eh_0'}
{'pid': 174, 'name': 'scsi_eh_1'}
{'pid': 175, 'name': 'scsi_eh_2'}
{'pid': 278, 'name': 'jbd2/sda3-8'}
{'pid': 279, 'name': 'ext4-dio-unwrit'}
{'pid': 364, 'name': 'udevd'}
{'pid': 415, 'name': 'bash'}
{'pid': 550, 'name': 'vmmemctl'}
{'pid': 604, 'name': 'flush-8:0'}
{'pid': 666, 'name': 'jbd2/sda1-8'}
{'pid': 667, 'name': 'ext4-dio-unwrit'}
{'pid': 721, 'name': 'kauditd'}
{'pid': 820, 'name': 'python'}
{'pid': 916, 'name': 'auditd'}
{'pid': 932, 'name': 'rsyslogd'}
{'pid': 981, 'name': 'sshd'}
{'pid': 1050, 'name': 'sshd'}
{'pid': 1252, 'name': 'bash'}
{'pid': 1328, 'name': 'master'}
{'pid': 1335, 'name': 'qmgr'}
{'pid': 1338, 'name': 'crond'}
{'pid': 1351, 'name': 'mingetty'}
{'pid': 1353, 'name': 'mingetty'}
{'pid': 1355, 'name': 'mingetty'}
{'pid': 1356, 'name': 'udevd'}
{'pid': 1358, 'name': 'udevd'}
{'pid': 1359, 'name': 'mingetty'}
{'pid': 1361, 'name': 'mingetty'}
{'pid': 1363, 'name': 'mingetty'}
{'pid': 1893, 'name': 'sshd'}
{'pid': 2038, 'name': 'bash'}
{'pid': 4015, 'name': 'httpd'}
{'pid': 4016, 'name': 'httpd'}
{'pid': 4017, 'name': 'httpd'}
{'pid': 4018, 'name': 'httpd'}
{'pid': 4019, 'name': 'httpd'}
{'pid': 4020, 'name': 'httpd'}
{'pid': 4021, 'name': 'httpd'}
{'pid': 4022, 'name': 'httpd'}
{'pid': 7487, 'name': 'httpd'}
{'pid': 7533, 'name': 'redis-server'}
{'pid': 9092, 'name': 'pickup'}
{'pid': 14423, 'name': 'httpd'}
{'pid': 15081, 'name': 'sshd'}
{'pid': 15426, 'name': 'bash'}
{'pid': 15948, 'name': 'redis-cli'}
{'pid': 22268, 'name': 'sshd'}
{'pid': 22392, 'name': 'bash'}
{'pid': 24213, 'name': 'httpd'}
{'pid': 24288, 'name': 'httpd'}
{'pid': 24302, 'name': 'httpd'}
{'pid': 25168, 'name': 'redis-server'}
{'pid': 25527, 'name': 'redis-server'}
{'pid': 25630, 'name': 'netcast-dashboard'}
{'pid': 26258, 'name': 'redis-server'}
{'pid': 26305, 'name': 'redis-sentinel'}
{'pid': 26330, 'name': 'sshd'}
{'pid': 26382, 'name': 'sftp-server'}
{'pid': 30660, 'name': 'mysqld_safe'}
{'pid': 30864, 'name': 'mysqld'}
{'pid': 32494, 'name': 'sshd'}
```

```
// gopsutil.go
package main

import (
        "fmt"

        "github.com/shirou/gopsutil/process"
)

func main() {
        pids, _ := process.Pids()
        for _, pid := range pids {
                p, _ := process.NewProcess(pid)
                name, _ := p.Name()
                fmt.Printf("{'pid':%d, 'name': %q}\n", pid, name)
        }
}
```

```
$ go run gopsutil.go
{'pid':1, 'name': "init"}
{'pid':2, 'name': "kthreadd"}
{'pid':3, 'name': "migration/0"}
{'pid':4, 'name': "ksoftirqd/0"}
{'pid':5, 'name': "migration/0"}
{'pid':6, 'name': "watchdog/0"}
{'pid':7, 'name': "migration/1"}
{'pid':8, 'name': "migration/1"}
{'pid':9, 'name': "ksoftirqd/1"}
{'pid':10, 'name': "watchdog/1"}
{'pid':11, 'name': "events/0"}
{'pid':12, 'name': "events/1"}
{'pid':13, 'name': "cgroup"}
{'pid':14, 'name': "khelper"}
{'pid':15, 'name': "netns"}
{'pid':16, 'name': "async/mgr"}
{'pid':17, 'name': "pm"}
{'pid':18, 'name': "sync_supers"}
{'pid':19, 'name': "bdi-default"}
{'pid':20, 'name': "kintegrityd/0"}
{'pid':21, 'name': "kintegrityd/1"}
{'pid':22, 'name': "kblockd/0"}
{'pid':23, 'name': "kblockd/1"}
{'pid':24, 'name': "kacpid"}
{'pid':25, 'name': "kacpi_notify"}
{'pid':26, 'name': "kacpi_hotplug"}
{'pid':27, 'name': "ata_aux"}
{'pid':28, 'name': "ata_sff/0"}
{'pid':29, 'name': "ata_sff/1"}
{'pid':30, 'name': "ksuspend_usbd"}
{'pid':31, 'name': "khubd"}
{'pid':32, 'name': "kseriod"}
{'pid':33, 'name': "md/0"}
{'pid':34, 'name': "md/1"}
{'pid':35, 'name': "md_misc/0"}
{'pid':36, 'name': "md_misc/1"}
{'pid':37, 'name': "linkwatch"}
{'pid':38, 'name': "khungtaskd"}
{'pid':39, 'name': "kswapd0"}
{'pid':40, 'name': "ksmd"}
{'pid':41, 'name': "khugepaged"}
{'pid':42, 'name': "aio/0"}
{'pid':43, 'name': "aio/1"}
{'pid':44, 'name': "crypto/0"}
{'pid':45, 'name': "crypto/1"}
{'pid':50, 'name': "kthrotld/0"}
{'pid':51, 'name': "kthrotld/1"}
{'pid':52, 'name': "pciehpd"}
{'pid':54, 'name': "kpsmoused"}
{'pid':55, 'name': "usbhid_resumer"}
{'pid':85, 'name': "kstriped"}
{'pid':169, 'name': "mpt_poll_0"}
{'pid':170, 'name': "mpt/0"}
{'pid':171, 'name': "scsi_eh_0"}
{'pid':174, 'name': "scsi_eh_1"}
{'pid':175, 'name': "scsi_eh_2"}
{'pid':278, 'name': "jbd2/sda3-8"}
{'pid':279, 'name': "ext4-dio-unwrit"}
{'pid':364, 'name': "udevd"}
{'pid':415, 'name': "bash"}
{'pid':550, 'name': "vmmemctl"}
{'pid':604, 'name': "flush-8:0"}
{'pid':666, 'name': "jbd2/sda1-8"}
{'pid':667, 'name': "ext4-dio-unwrit"}
{'pid':721, 'name': "kauditd"}
{'pid':916, 'name': "auditd"}
{'pid':932, 'name': "rsyslogd"}
{'pid':981, 'name': "sshd"}
{'pid':1050, 'name': "sshd"}
{'pid':1252, 'name': "bash"}
{'pid':1328, 'name': "master"}
{'pid':1335, 'name': "qmgr"}
{'pid':1338, 'name': "crond"}
{'pid':1351, 'name': "mingetty"}
{'pid':1353, 'name': "mingetty"}
{'pid':1355, 'name': "mingetty"}
{'pid':1356, 'name': "udevd"}
{'pid':1358, 'name': "udevd"}
{'pid':1359, 'name': "mingetty"}
{'pid':1361, 'name': "mingetty"}
{'pid':1363, 'name': "mingetty"}
{'pid':1893, 'name': "sshd"}
{'pid':2038, 'name': "bash"}
{'pid':3393, 'name': "go"}
{'pid':3460, 'name': "gopsutil"}
{'pid':4015, 'name': "httpd"}
{'pid':4016, 'name': "httpd"}
{'pid':4017, 'name': "httpd"}
{'pid':4018, 'name': "httpd"}
{'pid':4019, 'name': "httpd"}
{'pid':4020, 'name': "httpd"}
{'pid':4021, 'name': "httpd"}
{'pid':4022, 'name': "httpd"}
{'pid':7487, 'name': "httpd"}
{'pid':7533, 'name': "redis-server"}
{'pid':9092, 'name': "pickup"}
{'pid':14423, 'name': "httpd"}
{'pid':15081, 'name': "sshd"}
{'pid':15426, 'name': "bash"}
{'pid':15948, 'name': "redis-cli"}
{'pid':22268, 'name': "sshd"}
{'pid':22392, 'name': "bash"}
{'pid':24213, 'name': "httpd"}
{'pid':24288, 'name': "httpd"}
{'pid':24302, 'name': "httpd"}
{'pid':25168, 'name': "redis-server"}
{'pid':25527, 'name': "redis-server"}
{'pid':25630, 'name': "netcast-dashboa"}
{'pid':26258, 'name': "redis-server"}
{'pid':26305, 'name': "redis-sentinel"}
{'pid':26330, 'name': "sshd"}
{'pid':26382, 'name': "sftp-server"}
{'pid':30660, 'name': "mysqld_safe"}
{'pid':30864, 'name': "mysqld"}
{'pid':32494, 'name': "sshd"}
```

**Comment by jcaberio** [comment]
After reading the source code in [process_linux.go](https://github.com/shirou/gopsutil/blob/master/process/process_linux.go#L706), it turns out that the process name is derived from **/proc/$PID/status** which is the [root cause](http://askubuntu.com/questions/21650/can-i-use-standard-tools-to-get-the-full-name-of-a-process-when-its-name-has-em) of the truncated process name.

**Comment by jcaberio** [comment]
One solution is to use the same approach of python psutil described [here](https://github.com/giampaolo/psutil/blob/master/psutil/__init__.py#L637).

**Comment by shirou** [comment]
closed via #303. Thank you very much!
